code/datums/abilities/flock/flockmind.dm

```diff
@@ -311,17 +311,27 @@
 	name = "Fabricate Structure"
 	desc = "Create a structure tealprint for your drones to construct onto."
 	icon_state = "fabstructure"
-	cooldown = 4
+	cooldown = 4 SECONDS
 	targeted = 0
 
 /datum/targetable/flockmindAbility/createStructure/cast()
 	var/resourcecost = null
 	var/structurewantedtype = null
-	var/structurewanted = input("Select which structure you would like to create", "Tealprint Selection", "cancel") as null|anything in list("Collector")
+	var/turf/t = get_turf(holder.owner)
+	if(!istype(t, /turf/simulated/floor/feather))
+		boutput(holder.owner, "<span class='alert'>You aren't above a flocktile.</span>")//todo maybe make this flock themed?
+		return 1
+	if(locate(/obj/flock_structure) in t)
+		boutput(holder.owner, "<span class='alert'>There is already a flock structure on this flocktile!</span>")
+		return 1
+	var/structurewanted = input("Select which structure you would like to create", "Tealprint Selection", "cancel") as null|anything in list("Collector", "Sentinel")
```

Comment:
consider tgui list input

could even do a custom theme eventually

Reply:
yeah, i was planning to make the flock control panel (i forget the name of it) thing use tgui too. I just have to learn tgui first, ill do both in the future soon

Reply:
tgui list input requires no knowledge of tgui at all, you just call a different proc